pandas/core/ops/array_ops.py

```diff
@@ -321,6 +325,16 @@ def comparison_op(left: ArrayLike, right: Any, op) -> ArrayLike:
                 "Lengths must match to compare", lvalues.shape, rvalues.shape
             )
 
+    if (is_string_dtype(lvalues) and is_object_dtype(rvalues)) or (
+        is_object_dtype(lvalues) and is_string_dtype(rvalues)
```

Comment:
Checking for string dtype for an array _can_ be expensive in case the array is object dtype (at that point it will scan all values to check if they are strings). So we might want to try avoid that at this level. 
I _think_ we could handle the issue specifically for the ArrowExtensionArray itself (see the code I referenced in https://github.com/pandas-dev/pandas/issues/60228#issuecomment-2462533868)